### PR TITLE
feature/work-environments

### PR DESCRIPTION
**Pull request by seco-plr**

### CODE REVIEW

File: domain/src/main/java/ch/admin/seco/alv/service/competencecatalog/domain/workenvironment/WorkEnvironmentBuilder.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
+package ch.admin.seco.alv.service.competencecatalog.domain.workenvironment;
+
+import ch.admin.seco.alv.service.competencecatalog.domain.TranslatedString;
+
+public class WorkEnvironmentBuilder {
```
Comment (emina-str) [inline domain/src/main/java/ch/admin/seco/alv/service/competencecatalog/domain/workenvironment/WorkEnvironmentBuilder.java]: we don't do getters in Builder, only setters.

**Comment by emina-str** [comment]
just fix this small issue with Builder, otherwise from my side ok.

File: domain/src/main/java/ch/admin/seco/alv/service/competencecatalog/domain/workenvironment/WorkEnvironmentBuilder.java (hunk 1 above)
Comment (seco-plr) [reply domain/src/main/java/ch/admin/seco/alv/service/competencecatalog/domain/workenvironment/WorkEnvironmentBuilder.java]: Fair enough, fixed the Builder in 77de4b7aa41b100dfa795102cde8efc78969ffba

File: application/src/main/java/ch/admin/seco/alv/service/competencecatalog/application/SequenceType.java [hunk 2]
```diff
@@ -6,7 +6,10 @@
     KNOW_HOW("HLW"),
     CH_FICHE("CHF"),
     COMPETENCE_SET("KPZ"),
-    REQUIREMENT("VRS");
+    REQUIREMENT("VRS"),
+    PROFESSIONAL_ENVIRONMENT("ARU"),
+    SECTOR("BRN"),
+    WORK_CONDITION("ARB");
```
Comment (cjpsycos) [inline application/src/main/java/ch/admin/seco/alv/service/competencecatalog/application/SequenceType.java]: Defined is: ENVIRONMENT, SECTOR and CONDITION

File: domain/src/main/java/ch/admin/seco/alv/service/competencecatalog/domain/workenvironment/WorkEnvironmentType.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package ch.admin.seco.alv.service.competencecatalog.domain.workenvironment;
+
+public enum WorkEnvironmentType {
+
+    PROFESSIONAL_ENVIRONMENT,
+    SECTOR,
+    WORK_CONDITION
```
Comment (cjpsycos) [inline domain/src/main/java/ch/admin/seco/alv/service/competencecatalog/domain/workenvironment/WorkEnvironmentType.java]: Defined is: ENVIRONMENT, SECTOR and CONDITION

File: domain/src/main/java/ch/admin/seco/alv/service/competencecatalog/domain/workenvironment/events/WorkEnvironmentCreatedEvent.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package ch.admin.seco.alv.service.competencecatalog.domain.workenvironment.events;
+
+import ch.admin.seco.alv.service.competencecatalog.core.events.DomainEventType;
+import ch.admin.seco.alv.service.competencecatalog.domain.workenvironment.WorkEnvironment;
+
+public class WorkEnvironmentCreatedEvent extends WorkEnvironmentEvent {
+
+    public WorkEnvironmentCreatedEvent(WorkEnvironment workEnvironment) {
+        super(new DomainEventType("WorkEnvironmentCreatedEvent"), workEnvironment);
```
Comment (cjpsycos) [inline domain/src/main/java/ch/admin/seco/alv/service/competencecatalog/domain/workenvironment/events/WorkEnvironmentCreatedEvent.java]: pass class to domainEventType

File: infrastructure-web/src/main/resources/liquibase/changelog/20200115080000_added_work_environments.xml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+            <column name="DESCRIPTION_EN" type="VARCHAR(1024)"/>
+            <column name="DESCRIPTION_FR" type="VARCHAR(1024)"/>
+            <column name="DESCRIPTION_IT" type="VARCHAR(1024)"/>
+        </createTable>
+    </changeSet>
+
+    <changeSet author="capl" id="20200115080000-02">
+        <sql>CREATE SEQUENCE SEQ_PROFESSIONAL_ENVIRONMENT MINVALUE 1 MAXVALUE 99999999999 NO CYCLE</sql>
+        <sql>CREATE SEQUENCE SEQ_SECTOR MINVALUE 1 MAXVALUE 99999999999 NO CYCLE</sql>
+        <sql>CREATE SEQUENCE SEQ_WORK_CONDITION MINVALUE 1 MAXVALUE 99999999999 NO CYCLE</sql>
```
Comment (cjpsycos) [inline infrastructure-web/src/main/resources/liquibase/changelog/20200115080000_added_work_environments.xml]: Defined is: ENVIRONMENT, SECTOR and CONDITION

File: infrastructure-web/src/main/resources/liquibase/master.xml [hunk 6]
```diff
@@ -4,13 +4,20 @@
...
+    <include file="classpath:/liquibase/changelog/20191011083000_added_event_store.xml"
+             relativeToChangelogFile="false"/>
+    <include file="classpath:/liquibase/changelog/20191211080000_add_column_cc_objects.xml"
+             relativeToChangelogFile="false"/>
+    <include file="classpath:/liquibase/changelog/20191211081000_added_spring_jdbc_integration.xml"
+             relativeToChangelogFile="false"/>
     <include file="classpath:/liquibase/changelog/20200109080000_added_requirements.xml"
              relativeToChangelogFile="false"/>
+    <include file="classpath:/liquibase/changelog/20200115080000_added_work_environments.xml"
+             relativeToChangelogFile="false"/>
```
Comment (cjpsycos) [inline infrastructure-web/src/main/resources/liquibase/master.xml]: you should use the common style settings for the IDE

File: infrastructure/src/test/resources/schema-h2.sql [hunk 7]
```diff
@@ -3,4 +3,7 @@ CREATE SEQUENCE SEQ_COMPETENCE_SET MINVALUE 1 MAXVALUE 99999999999 NO CYCLE;
 CREATE SEQUENCE SEQ_KNOW_HOW MINVALUE 1 MAXVALUE 99999999999 NO CYCLE;
 CREATE SEQUENCE SEQ_KNOW_HOW_INDICATOR MINVALUE 1 MAXVALUE 99999999999 NO CYCLE;
 CREATE SEQUENCE SEQ_KNOWLEDGE MINVALUE 1 MAXVALUE 99999999999 NO CYCLE;
-CREATE SEQUENCE SEQ_REQUIREMENT MINVALUE 1 MAXVALUE 99999999999 NO CYCLE
\ No newline at end of file
+CREATE SEQUENCE SEQ_REQUIREMENT MINVALUE 1 MAXVALUE 99999999999 NO CYCLE;
+CREATE SEQUENCE SEQ_PROFESSIONAL_ENVIRONMENT MINVALUE 1 MAXVALUE 99999999999 NO CYCLE;
+CREATE SEQUENCE SEQ_SECTOR MINVALUE 1 MAXVALUE 99999999999 NO CYCLE;
+CREATE SEQUENCE SEQ_WORK_CONDITION MINVALUE 1 MAXVALUE 99999999999 NO CYCLE;
```
Comment (cjpsycos) [inline infrastructure/src/test/resources/schema-h2.sql]: Defined is: ENVIRONMENT, SECTOR and CONDITION

**Pull request merged**